ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -1,14 +1,45 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
 import { PageSection, Title, Flex, FlexItem } from '@patternfly/react-core';
+import { Model } from '@patternfly/react-topology';
 
+import { fetchNetworkFlowGraph } from 'services/NetworkService';
+import { fetchClustersAsArray, Cluster } from 'services/ClustersService';
 import PageTitle from 'Components/PageTitle';
 import NetworkGraph from './NetworkGraph';
+import { transformData, graphModel } from './utils';
 
 import './NetworkGraphPage.css';
 
 function NetworkGraphPage() {
     const { detailType, detailId } = useParams();
+    const [model, setModel] = useState<Model>({
+        graph: graphModel,
+    });
+    const [clusters, setClusters] = useState<Cluster[]>([]);
+
+    useEffect(() => {
+        fetchClustersAsArray()
+            .then((response) => {
+                setClusters(response);
+            })
+            .catch(() => {
+                // TODO
+            });
+    }, []);
```

Comment:
I had a draft PR, just to test out cluster select, that was using @sachaudh 's suggestion of a custom hook for this.
https://github.com/stackrox/stackrox/pull/3122/files#diff-3117d6b9291b82097bae968b466a3a827737a027595aefde5378d1b2d1a59fca

Once this PR of yours is merged, I will "merge" your data flow changes with the select logic, in a fresh branch, so everybody can share the cluster-fetching logic.

Kum-ba-ya.